pulp_container/app/registry_api.py

```diff
@@ -564,18 +564,95 @@ def create(self, request, path):
         """
         Create a new upload.
 
-        Note: We do not support monolithic upload.
         """
         _, repository = self.get_dr_push(request, path, create=True)
 
         if self.tries_to_mount_blob(request):
             response = self.mount_blob(request, path, repository)
+            return response
+        elif digest := request.query_params.get("digest"):
+            # if the digest parameter is present, the request body will be
+            # used to complete the upload in a single request.
+            self.single_request_upload(request, path, repository, digest)
         else:
             upload = models.Upload(repository=repository, size=0)
             upload.save()
             response = UploadResponse(upload=upload, path=path, request=request)
+            return response
 
-        return response
+    def create_single_chunk_artifact(self, chunk):
```

Comment:
I miss basic error handling in this method (e.g., digest invalid, omitted content-type, etc.).

Reply:
Digest validation is done in `init_and_validate`.

Reply:
There are few places that would benefit from digest validation format, at least in the PUT request.
I would not bother with that, it would just give a 404. Or at least not in the scope of this PR.